@here/harp-mapview/lib/text/TextElementState.ts

```diff
@@ -0,0 +1,231 @@
+            this.state === FadingState.FadedIn ||
+            this.state === FadingState.FadingOut;
+        return visible;
+    }
+}
+
+/**
+ * `TextElementState` keeps the current state of a text element while it's being rendered.
+ */
+export class TextElementState {
+    /**
+     * @hidden
+     * Used during label placement to reserve space from front to back.
+     */
+    m_viewDistance?: number;
+
+    /**
+     * @hidden
+     * Used during rendering. The array type is used for line markers only, which have a points
+     * array and multiple icon positions to render. Since line markers use the same renderState
+     * for text part and icon, there is no separate array of [[RenderState]]s for the text parts
+     * of the line markers.
+     */
+    m_iconRenderStates?: RenderState | RenderState[];
+
+    /**
+     * @hidden
+     * Used during rendering.
+     */
+    m_textRenderState?: RenderState;
+
+    /**
+     * Initializes the state.
+     * @param textElementType The text element type.
+     * @param disableFading True if fading is disabled, false otherwise.
+     * @param textElementPointCount Number of points the text element has.
+     */
+    initialize(
+        textElementType: TextElementType,
+        disableFading: boolean,
+        textElementPointCount: number = 1
+    ) {
+        const fadingTime = disableFading === true ? 0 : DEFAULT_FADE_TIME;
+
+        if (textElementType === TextElementType.LineMarker) {
+            this.m_iconRenderStates = new Array<RenderState>();
+            for (let i = 0; i < textElementPointCount; ++i) {
+                const iconRenderStates = this.m_iconRenderStates as RenderState[];
+                const renderState = new RenderState();
+                renderState.state = FadingState.FadingIn;
+                renderState.fadingTime = fadingTime;
+                iconRenderStates.push(renderState);
+            }
+            return;
+        }
+
+        this.m_textRenderState = new RenderState();
+        this.m_textRenderState.fadingTime = fadingTime;
+
+        if (textElementType === TextElementType.PoiLabel) {
+            this.m_iconRenderStates = new RenderState();
+            this.m_iconRenderStates.fadingTime = fadingTime;
+        }
+    }
+
+    get initialized(): boolean {
+        return this.m_textRenderState !== undefined || this.m_iconRenderStates !== undefined;
+    }
+
+    /**
+     * Clears the state.
+     */
+    clear() {
+        this.m_iconRenderStates = undefined;
+        this.m_textRenderState = undefined;
+    }
+
+    /**
+     * Returns the last computed distance of the text element to the camera.
+     * @returns Distance to camera.
+     */
+    get viewDistance(): number | undefined {
+        return this.m_viewDistance;
+    }
+
+    set viewDistance(distance: number | undefined) {
+        this.m_viewDistance = distance;
+    }
+
+    /**
+     * Returns the text render state.
+     * @returns The text render state.
```

Comment:
only duplicate if it brings value, otherwise it is just a waste / maintenance effort.

Reply:
done